File: go/muse/algorithms/search/search_test.go
package search

import (
	"testing"

	"muse/algorithms/search/tests"
)

func TestBinarySearch(t *testing.T) {
	t.Run("BinarySearch", tests.DeriveIncreasing(BinarySearch, 100000))
}

func TestLinearSearch(t *testing.T) {
	t.Run("LinearSearch", tests.DeriveIncreasing(LinearSearch, 10000))
}

File: python/muse/algorithms/search/search_test.py
from muse.algorithms import search
from muse.algorithms.search import tests
from muse.util import test_runner


def test_binary_search() -> bool:
    return tests.derive_increasing(search.binarysearch, 100000)()


def test_linear_search() -> bool:
    return tests.derive_increasing(search.linearsearch, 10000)()


def main() -> None:
    test_runner.pick(test_binary_search)

    test_runner.pick(test_linear_search)


if __name__ == "__main__":
    main()
